### PR TITLE
Integration test for iris.plot.contour with 2d coords.

### LINKED ISSUE #3167: Add tests for 2d coords contouring
At present (https://github.com/SciTools/iris/tree/5923aaa8e725831a1575678fd93c938bfad36e95), we test :
  * various tests for 2d coord handling functions in tests.unit.plot : `test__get_plot_defn[_custom_coords_picked]`, `test__check_bounds_contiguity_and_mask`
  * integration tests for pcolormesh in tests.integration.plot : `test_plot_2d_coords`.

We ought to ***also*** have an integration test proving 2d coords operation in contour/contourf 
Probably also belongs in  `test_plot_2d_coords`

### PR DESCRIPTION
**Pull request by pp-mo**
Needs https://github.com/SciTools/test-iris-imagehash/pull/21

Closes #3167

### CODE REVIEW

File: lib/iris/tests/integration/plot/test_plot_2d_coords.py [hunk 1]
```diff
@@ -26,10 +26,16 @@
 # importing anything else
 import iris.tests as tests
 
-import cartopy.crs as ccrs
+import numpy as np
```
Comment (dkillick) [inline lib/iris/tests/integration/plot/test_plot_2d_coords.py]: @pp-mo why is the `NumPy` import separate at the top? At minimum I feel it should be after the `pyplot` import. Preferably they would both be after the Iris imports.

File: lib/iris/tests/integration/plot/test_plot_2d_coords.py [hunk 2]
```diff
@@ -63,5 +69,29 @@ def test_2d_coord_bounds_northpolarstereo(self):
         self.check_graphic()
 
 
+@tests.skip_plot
+@tests.skip_data
```
Comment (dkillick) [inline lib/iris/tests/integration/plot/test_plot_2d_coords.py]: @pp-mo you shouldn't need this decorator as you're not making use of external test data.

Comment (pp-mo) [reply lib/iris/tests/integration/plot/test_plot_2d_coords.py]: :+1: cut+paste error, I think.
Good spot !!

File: lib/iris/tests/integration/plot/test_plot_2d_coords.py (hunk 1 above)
Comment (pp-mo) [reply lib/iris/tests/integration/plot/test_plot_2d_coords.py]: Presumably you mean "NumPy ... after the pyplot ... ***before*** the Iris imports" ??

My understanding of our guidelines for imports is to separate into 3 or 4 specific groups : 
  1. standard-library packages
  2. other 3rd party libraries
  3. "own" packages + modules
  4. (in tests only) test subjects

So here, numpy and pyplot should be in the same block, 
but before any Iris stuff 
(except iris.tests, which is required to happen first. :nauseated_face: yech ).

This agree with your understanding @dkillick ?

Comment (dkillick) [reply lib/iris/tests/integration/plot/test_plot_2d_coords.py]: > This agree with your understanding

Yes... and no, at the same time! Basically, I feel like we've slipped away from separating (2.) and (3.), even in test modules. It's not worth sweating about here, but I would still like to see the `NumPy` import at least after the `pyplot` import 😉

File: lib/iris/tests/integration/plot/test_plot_2d_coords.py (hunk 2 above)
Comment (dkillick) [reply lib/iris/tests/integration/plot/test_plot_2d_coords.py]: Ah yes, that old chestnut 🌰

File: lib/iris/tests/integration/plot/test_plot_2d_coords.py (hunk 1 above)
Comment (pp-mo) [reply lib/iris/tests/integration/plot/test_plot_2d_coords.py]: :+1:

**Comment by dkillick** [comment]
Whoops, one of the Travis jobs had an error. I've restarted it, whcih hopefully that will solve the problem.

**Pull request merged**